src/funcs.rs

```diff
@@ -0,0 +1,237 @@
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fmt;
+use std::fmt::{Debug, Formatter};
+
+use chrono::{DateTime, FixedOffset};
+use itertools::Itertools;
+use lazy_static::lazy_static;
+
+use crate::data;
+use crate::operator::EvalError;
+
+/// Enum used to capture a static function that can be called by the expression language.
+#[derive(Clone, Copy)]
+pub enum FunctionWrapper {
+    Float1(fn(f64) -> f64),
+    Float2(fn(f64, f64) -> f64),
+    String1(fn(&str) -> Result<data::Value, EvalError>),
+    String2(fn(&str, &str) -> Result<data::Value, EvalError>),
+    Generic(fn(&Vec<data::Value>) -> Result<data::Value, EvalError>),
+}
+
+/// Struct used to capture the name of the function in the expression language and a pointer
+/// to the implementation.
+#[derive(Clone, Copy)]
+pub struct FunctionContainer {
+    name: &'static str,
+    func: FunctionWrapper,
+}
+
+impl Debug for FunctionContainer {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(self.name)
+    }
+}
+
+impl FunctionContainer {
+    pub fn new(name: &'static str, func: FunctionWrapper) -> Self {
+        FunctionContainer { name, func }
+    }
+
+    pub fn eval_func(&self, args: &Vec<data::Value>) -> Result<data::Value, EvalError> {
+        match self.func {
+            FunctionWrapper::Float1(func) => match args.as_slice() {
+                [data::Value::Float(fl)] => Ok(data::Value::from_float(func(fl.0))),
```

Comment:
I guess it might be a tiny bit faster, but I think we might as well only have the `[arg0]` case

Reply:
I'm not sure what you meant here.